src/main/scala/org/constellation/gossip/sampling/RandomPeerSampling.scala

```diff
@@ -11,10 +15,28 @@ import scala.util.Random
 /**
   * Peer sampling strategy based on uniform random sample selection
   */
-class RandomPeerSampling[F[_]: Concurrent](selfId: Id, cluster: Cluster[F]) extends PeerSampling[F, Id] {
-
-  def selectPaths(fanout: Int): F[Seq[IndexedSeq[Id]]] =
-    getPeers.map(randomPath(_, fanout))
+class RandomPeerSampling[F[_]](selfId: Id, cluster: Cluster[F])(implicit F: Concurrent[F]) extends PeerSampling[F] {
+
+  private val logger: SelfAwareStructuredLogger[F] = Slf4jLogger.getLogger[F]
+
+  def selectPaths(fanout: Int): F[List[GossipPath]] =
```

Comment:
We can just remove this or re-purpose this method to flatten the partition map

Reply:
`RandomPeerSampling` is just a sample implementation of random path selection. It won't be used in production because we will use your partitioner instead.

Basically as you can see we extend here `trait PeerSampling` which forces you to implement `def selectPaths: F[List[GossipPath]]`. We just need implementation of `PeerSampling` which uses partitioner under the hood and it will work out of the box.